### PR TITLE
[MS] Use code to validation new email

### LINKED ISSUE #7794: [Client] Improved email update modal process (`PersonalDataPage`)
https://www.figma.com/design/8WXUivEfT1x4FTzCr3WNxd/Parsec-Client?node-id=118-6764&t=LpdWcegpKK301cc0-11

### PR DESCRIPTION
**Pull request by Max-7**
Closes #7794 

- [X] Keep changes in the pull request as small as possible
- [X] Ensure the commit history is sanitized
- [X] Give a meaningful title to your PR
- [X] Describe your changes
- [X] Link any related issue in the description

### CODE REVIEW

File: client/src/views/client-area/personal-data/AuthenticationModal.vue [hunk 1]
```diff
@@ -65,32 +143,112 @@ const newEmailValidator: IValidator = async function (value: string) {
...
 });
 
 async function submit(): Promise<boolean> {
-  if (!isFormValid()) {
-    return false;
+  querying.value = true;
+  if (step.value === Steps.NewEmail) {
+    await switchStep(Steps.Password);
+  } else if (step.value === Steps.Password) {
+    const response = await BmsAccessInstance.get().updateEmailSendCode(
```
[the remark this answers is not in the document]
Comment (Max-7) [reply client/src/views/client-area/personal-data/AuthenticationModal.vue:155]: The backend does not handle codes yet (https://github.com/Scille/parsec-bms/pull/998)

File: client/src/views/client-area/personal-data/AuthenticationModal.vue [hunk 2]
```diff
@@ -65,32 +143,112 @@ const newEmailValidator: IValidator = async function (value: string) {
...
+    const response = await BmsAccessInstance.get().updateEmail(
+      newEmailRef.value,
+      passwordRef.value,
+      validationCode.value.join(''),
+      longLocaleCodeToShort(I18n.getLocale()) as BmsLang,
+    );
+    if (response.isError) {
+      error.value = 'globalErrors.unexpected';
+      switch (response.status) {
+        case 400:
```
[the remark this answers is not in the document]
Comment (Max-7) [reply client/src/views/client-area/personal-data/AuthenticationModal.vue:174]: Backend doesn't implement the send code yet, errors should change (https://github.com/Scille/parsec-bms/pull/998/files#diff-dc4e527b178f074bc19fcd75f51f295b9ed8c67c22782648306b69c4837135c6R27)

File: client/src/views/client-area/personal-data/AuthenticationModal.vue [hunk 3]
```diff
@@ -65,32 +143,113 @@ const newEmailValidator: IValidator = async function (value: string) {
...
+      };
+      break;
+  }
+}
+
+async function isValidCode(code: Array<string>): Promise<boolean> {
+  return code.length === 6;
+}
+
+async function resendCode(): Promise<void> {
```
Comment (Ironicbay) [inline client/src/views/client-area/personal-data/AuthenticationModal.vue:237]: This should empty the code fields

File: client/src/views/client-area/personal-data/AuthenticationModal.vue [hunk 4]
```diff
@@ -3,31 +3,68 @@
...
+            @on-enter-keyup="submit"
+            :label="`${translationPrefix}.password`"
+            ref="passwordInput"
+          />
+        </div>
+        <div
+          class="code-container"
+          v-show="step === Steps.Code"
+        >
+          <ms-code-validation-input
```
Comment (Ironicbay) [inline client/src/views/client-area/personal-data/AuthenticationModal.vue:46]: This is not focused as the modal opens. I guess the component needs some tweaking in megashark-lib, currently the focus is done 200ms after `mount`.

**Review by Ironicbay — COMMENTED** [review]
When inputing an incorrect password then a correct code, issues occur after re-inputing the password and getting to the code validation a second time:
- The code fields are still filled with the previous code, but another has been sent so it's invalid. Either the code should stay valid if we update the password, or the fields have to be cleared
- Once an incorrect code is tried by the user, the button `Validate and update` stays disabled no matter what code the user inputs
- Code validation modal displays a green text `The code is valid` when the 6 fields are filled, regardless of the numbers

**Review by Ironicbay — APPROVED** [review]

**Pull request merged**